source/libraries/Contracts.ts

```diff
@@ -0,0 +1,53 @@
+import { Abi } from 'ethereum';
+import { CompilerOutput } from 'solc';
+
+export class Contract {
+    public readonly relativeFilePath: string;
+    public readonly contractName: string;
+    public readonly abi: Abi;
+    public readonly bytecode: Buffer;
+    public address?: string;
+
+    public constructor(relativeFilePath: string, contractName: string, abi: Abi, bytecode: Buffer) {
+        this.relativeFilePath = relativeFilePath;
+        this.contractName = contractName;
+        this.abi = abi;
+        this.bytecode = bytecode;
+    }
+}
+
+export class Contracts implements Iterable<Contract> {
+    private readonly contracts = new Map<string, Contract>();
+
+    public constructor(compilerOutput: CompilerOutput) {
+        for (let relativeFilePath in compilerOutput.contracts) {
+            for (let contractName in compilerOutput.contracts[relativeFilePath]) {
+                // don't include helper libraries
+                if (!relativeFilePath.endsWith(`${contractName}.sol`)) continue;
+                const abi = compilerOutput.contracts[relativeFilePath][contractName].abi;
+                if (abi === undefined) continue;
+                const bytecodeString = compilerOutput.contracts[relativeFilePath][contractName].evm.bytecode.object;
+                if (bytecodeString === undefined) continue;
+                // don't include interfaces
```

Comment:
This is probably something we should be doing in the python tests as well but I think it would be nice to throw here if the contract name doesn't begin with 'I' followed by a capital character. I've been bit somewhat frequently by non-interface contracts falling into this case from not properly fulfilling abstracts/interfaces.

Reply:
https://app.clubhouse.io/augur/story/4631/

Per discussion, I like this idea but would require changes to the contracts to apply naming convention for abstract contracts everywhere.  Implementation at that point is quite simple:
```
if (bytecodeString.length === 0 && !contractName.startsWith('I')) throw new Error(...);
```

Putting off to future PR.